File: api_gestions_des_depense_des_enfants/src/Models/Enfant.model.js
import mongoose from "mongoose";
import ParentModel from "./Parent.model.js";
const { Schema, model, models } = mongoose;

const enfantSchema = new Schema({
  nom: { type: String, required: true },
  CNE: { type: String, unique: true },
  image: { type: String, default: "enfant-default.png" },
  genre: { type: String, enum: ["Male", "Female"] },
  ecole: String,
  niveau: String,
  parent: { type: mongoose.Schema.Types.ObjectId, ref: "Parent" }, // Référence au parent
});

enfantSchema.pre("save", async function (next) {
  if (this.isModified("parent")) {
    const parent = await ParentModel.findById(this.parent);
    if (!parent) {
      throw new Error("Parent not found");
    }
    parent.enfants = [...parent.enfants, this._id];
    await parent.save();
  }
  next();
});

export default models.Enfant || model("Enfant", enfantSchema);

File: gestions_des_depense_des_enfants/lib/models/enfant.model.dart
import 'package:gestions_des_depense_des_enfants/models/parent.model.dart';

class EnfantModel {
  String? id;
  String? fullname;
  String? cne;
  String? image;
  String? genre;
  String? ecole;
  String? niveau;
  ParentModel? parent;

  EnfantModel({
    this.id,
    this.fullname,
    this.cne,
    this.image,
    this.genre,
    this.ecole,
    this.niveau,
    this.parent,
  });

  factory EnfantModel.fromJson(Map<String, dynamic> json) {
    return EnfantModel(
      id: json['_id'],
      fullname: json['nom'],
      cne: json['CNE'],
      image: json['image'],
      genre: json['genre'],
      ecole: json['ecole'],
      niveau: json['niveau'],
      parent: ParentModel.fromJson(json['parent']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'fullname': fullname,
      'cne': cne,
      'image': image,
      'ecole': ecole,
      'niveau': niveau,
      'parent': parent?.toJson(),
    };
  }
}
